subsys/bluetooth/shell/bt.c

```diff
@@ -1970,15 +1969,18 @@ static int cmd_adv_data(const struct shell *sh, size_t argc, char *argv[])
 	atomic_set_bit_to(adv_set_opt[selected_adv], SHELL_ADV_OPT_APPEARANCE,
 			  appearance);
 
-	ad_len = ad_init(&ad[*data_len], ARRAY_SIZE(ad) - *data_len,
-			 adv_set_opt[selected_adv]);
-	if (ad_len < 0) {
+	len = ad_init(&data[*data_len], ARRAY_SIZE(ad) - *data_len, adv_set_opt[selected_adv]);
+	if (len < 0) {
 		shell_error(sh, "Failed to initialize stack advertising data");
 
 		return -ENOEXEC;
 	}
 
-	ad_len += *data_len;
+	if (data == ad) {
+		ad_len += len;
+	} else {
+		sd_len += len;
+	}
```

Comment:
How does this work in practice actually? 

Can we set `ad` and `sd` in a single command? And if so, then I guess it's the latter that will have the data from `ad_init`?

Reply:
> How does this work in practice actually?
> 
> Can we set `ad` and `sd` in a single command? And if so, then I guess it's the latter that will have the data from `ad_init`?

We can only set `ad` and `sd` separately .  `sd` can get data from `ad_init`.

Reply:
> We can only set ad and sd separately . sd can get data from ad_init.

Okay, so when `bt_le_ext_adv_set_data` is called, we either set `sd` or `ad`, but never both? So the user cannot do a mix of `ad` and `sd`?

Reply:
Seems only legecy and scannable and adv can set both `sd` and `ad` when `bt_le_ext_adv_set_data` called.
But we couldn't do that in `cmd_adv_data`.   It‘s not introduced by this PR.

This PR mainly want to fix https://github.com/zephyrproject-rtos/zephyr/issues/62235